schematic/schemas/json_schema_component_generator.py

```diff
@@ -0,0 +1,250 @@
+        """
+        Execute the steps to generate the JSON schema for a single component.
+        """
+        # parse model in cases where a component was provided and it wasn't necessary before
+        if not self.parsed_model:
+            self._parse_model()
+
+        # Direct the generation of the jsonschema for a single component
+        generator = JsonSchemaComponentGenerator(
+            data_model=self.data_model,
+            component=component,
+            output_directory=self.output_directory,
+            parsed_model=self.parsed_model,
+        )
+
+        generator.get_component_json_schema()
+        generator.add_description_to_json_schema()
+        generator.write_json_schema_to_file()
+
+        return generator.component_json_schema
+
+
+class JsonSchemaComponentGenerator:
+    """
+    Used to generate a jsonschema for a given component and write it to a file.
+    """
+
+    def __init__(
+        self,
+        data_model: str,
+        component: str,
+        output_directory: Path,
+        parsed_model: Dict[str, Any],
+    ):
+        self.data_model = data_model
+        self.parsed_model = parsed_model
+        self._get_data_model_graph_explorer()
+
+        if component is None:
+            raise ValueError("A component must be provided")
+
+        # the component can be provided as either a class label or display name
+        # internally all the work is done with the class label
+        component_class_label = self.dmge.get_node_label(component)
+        self.component = component_class_label if component_class_label else component
+
+        self.output_path = self._build_output_path(output_directory)
+
+        return
+
+    def _build_output_path(self, output_directory: Path) -> Path:
+        """
+        Build the output path for the JSON schema file.
+        Args:
+            output_directory: The directory where the JSON schema file will be saved.
+        Returns:
+            output_path: The path to the JSON schema file.
+        """
+
+        stripped_component = self.component.replace(" ", "")
+        return Path(output_directory, f"{stripped_component}_validation_schema.json")
+
+    def _get_data_model_graph_explorer(
+        self,
+    ) -> None:
+        """'
+        Create a DataModelGraphExplorer object to explore the data model graph.
+        """
+        # Instantiate DataModelGraph
+        data_model_grapher = DataModelGraph(self.parsed_model, "class_label")
+
+        # Generate graph
+        graph_data_model = data_model_grapher.graph
+
+        self.dmge = DataModelGraphExplorer(graph_data_model)
+
+        return
+
+    def get_component_json_schema(
+        self,
+    ) -> None:
+        """
+        Generate the JSON schema for the component using the DataModelJSONSchema instance.
+        The jsonschema is incomplete and will lack the description of the component.
+        """
+        schema_name = self.component + "_validation"
```

Comment:
currently the file paths are organized by component names which makes sense if user uses one schema. But what if the user uses two schemas? and those two schemas have the same components?